bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/NibeGW/NibeGW.ino

```diff
@@ -234,24 +245,37 @@ void nibeCallbackMsgReceived(const byte* const data, int len)
 
 int nibeCallbackTokenReceived(eTokenType token, byte* data)
 {
+  int len = 0;
   if (ethernetInitialized)
   {
     if (token == READ_TOKEN)
     {
       DEBUG_PRINT(2, "Read token received\n");
-      int packetSize = udp.parsePacket();
-      if (packetSize)
-        return udp.read(data, packetSize);
+      int packetSize = udp4readCmnds.parsePacket();
+      if (packetSize) {
+        len = udp4readCmnds.read(data, packetSize);
+        udp4readCmnds.flush();
```

Comment:
Flush applies only for output, right? So why it's needed here?

Reply:
You are right. It is not required for Ethernet library and only required for UIPEthernet (ENC28J60).

Reply:
Fixed it.